sdk/sdk/src/main/java/org/apache/carbondata/sdk/file/CarbonIUD.java

```diff
@@ -0,0 +1,376 @@
+   * @param column    is the columnName on which records have to be updated
+   * @param value     of column on which the records have to be updated
+   * @param updColumn is the name of updatedColumn
+   * @param updValue  is the value of updatedColumn
+   * @return CarbonUID
+   */
+  public CarbonIUD update(String path, String column, String value, String updColumn,
+      String updValue) {
+    prepareUpdate(path, column, value, updColumn, updValue);
+    return this;
+  }
+
+  /**
+   * This method updates the rows at given path by applying the filterExpression
+   *
+   * @param path                        is the table path on which update is performed.
+   * @param filterExpression            is the expression object to update the records
+   * @param updatedColumnToValueMapping contains the mapping of updatedColumns to updatedValues
+   * @throws IOException
+   * @throws InterruptedException
+   * @throws InvalidLoadOptionException
+   */
+  public void update(String path, Expression filterExpression,
+      Map<String, String> updatedColumnToValueMapping)
+      throws IOException, InterruptedException, InvalidLoadOptionException {
+    File[] indexFiles = getCarbonIndexFile(path);
+    Schema schema = CarbonSchemaReader.readSchema(indexFiles[0].getAbsolutePath()).asOriginOrder();
+    Field[] fields = schema.getFields();
+    String[] projectionColumns = new String[fields.length + 1];
+    for (int i = 0; i < fields.length; i++) {
+      projectionColumns[i] = (fields[i].getFieldName());
+    }
+    projectionColumns[projectionColumns.length - 1] =
+        CarbonCommonConstants.CARBON_IMPLICIT_COLUMN_TUPLEID;
+    CarbonWriter writer =
+        CarbonWriter.builder().outputPath(path).withCsvInput(schema).writtenBy("CarbonIUD").build();
+    CarbonReader reader =
+        CarbonReader.builder(path).projection(projectionColumns).filter(filterExpression).build();
+    RecordWriter<NullWritable, ObjectArrayWritable> deleteDeltaWriter =
+        CarbonTableOutputFormat.getDeleteDeltaRecordWriter(path);
+    ObjectArrayWritable writable = new ObjectArrayWritable();
+
+    while (reader.hasNext()) {
+      Object[] row = (Object[]) reader.readNextRow();
+      writable.set(Arrays.copyOfRange(row, row.length - 1, row.length));
+      for (Map.Entry<String, String> column : updatedColumnToValueMapping.entrySet()) {
+        row[getColumnIndex(fields, column.getKey())] = column.getValue();
+      }
+      writer.write(Arrays.copyOfRange(row, 0, row.length - 1));
+      deleteDeltaWriter.write(NullWritable.get(), writable);
+    }
+    deleteDeltaWriter.close(null);
+    writer.close();
+    reader.close();
+  }
+
+  /**
+   * Calling this method will start the execution of update process
+   *
+   * @throws IOException
+   * @throws InterruptedException
+   * @throws InvalidLoadOptionException
+   */
+  public void closeUpdate() throws IOException, InterruptedException, InvalidLoadOptionException {
```

Comment:
No need to expose closeUpdate and closeDelete to the user, these can be private. The user can simply call commit on all the operations and commit will handle internally

Reply:
Done